app/models/course/material/folder.rb

```diff
@@ -10,6 +10,12 @@ class Course::Material::Folder < ActiveRecord::Base
   # TODO: Remove this after schema_validations #21 was closed
   validates :name, uniqueness: { case_sensitive: false, scope: :parent_id }, if: :parent
 
+  def files=(files)
```

Comment:
folders do have many files, roughly equivalent to folder has_many :files, through: :materials. what's wrong with that?